util/util.go

```diff
@@ -20,12 +22,100 @@ import (
 var (
 	MaximumVolumeNameSize = 64
 	parsePattern          = regexp.MustCompile(`(.*):(\d+)`)
+	Logrotator            *lumberjack.Logger
 )
 
 const (
+	LogInfo        = "log.info"
 	BlockSizeLinux = 512
+
+	MaxLogFileSize         = "maxLogFileSize"
+	RetentionPeriod        = "retentionPeriod"
+	MaxBackups             = "maxBackups"
+	DefaultLogFileSize     = 100
+	DefaultRetentionPeriod = 180
+	DefaultMaxBackups      = 5
 )
 
+type LogToFile struct {
+	Enable          bool `json:"enable"`
+	MaxLogFileSize  int  `json:"maxlogfilesize"`
+	RetentionPeriod int  `json:"retentionperiod"`
+	MaxBackups      int  `json:"maxbackups"`
+}
+
+func StartLoggingToFile(dir string, lf LogToFile) error {
+	fileName := dir + "/replica.log"
+	Logrotator = &lumberjack.Logger{
+		Filename:   fileName,
+		MaxSize:    lf.MaxLogFileSize,
+		MaxAge:     lf.RetentionPeriod,
+		MaxBackups: lf.MaxBackups,
+		LocalTime:  true,
+	}
+	logrus.SetOutput(io.MultiWriter(os.Stderr, Logrotator))
+	if err := WriteLogInfo(dir, lf); err != nil {
+		return err
+	}
+	logrus.Infof("Configured logging with retentionPeriod: %v, maxLogFileSize: %v, maxBackups: %v",
+		lf.RetentionPeriod, lf.MaxLogFileSize, lf.MaxBackups)
+	return nil
+}
+
+func ReadLogInfo(dir string) (LogToFile, error) {
+	lf := &LogToFile{}
+	p := dir + "/" + LogInfo
+	f, err := os.Open(p)
+	if err != nil {
+		return LogToFile{}, err
+	}
+	defer f.Close()
+
+	err = json.NewDecoder(f).Decode(lf)
+	return *lf, err
+}
+
+func WriteLogInfo(dir string, lf LogToFile) error {
+	path := dir + "/" + LogInfo
+	f, err := os.Create(path + ".tmp")
+	if err != nil {
+		logrus.Errorf("failed to create temp file: %s while WriteLogInfo", LogInfo)
+		return err
+	}
+	defer f.Close()
+
+	if err := json.NewEncoder(f).Encode(&lf); err != nil {
+		logrus.Errorf("failed to encode the data to file: %s", f.Name())
+		return err
+	}
+
+	if err := f.Close(); err != nil {
+		logrus.Errorf("failed to close file after encoding to file: %s", f.Name())
+		return err
+	}
+
+	return os.Rename(path+".tmp", path)
+}
+
+func SetLogging(dir string, lf LogToFile) error {
+	// close existing one if already configured
+	if Logrotator != nil {
+		if err := Logrotator.Close(); err != nil {
+			return err
+		}
+	}
+
+	if !lf.Enable {
+		logrus.Infof("Disable logging to file")
+		return WriteLogInfo(dir, lf)
+	}
+	return StartLoggingToFile(dir, lf)
+}
+
+func LogRotate(dir string) error {
```

Comment:
this function is not used any where ?

Reply:
Keeping it here for future usage, like if user wants to trigger logrotation without changing the existing policy.